url3f8d6440dd_huiwq1990_CognativeModel_tgz-pJ8-clarion_samples_HelloWorldJ8/src/clarion/system/MS.java

```diff
@@ -0,0 +1,805 @@
+ * <ul>
+ * <li>The Behavioral Approach System - for those drives that are approach in nature.*</li>
+ * <li>The Behavioral Inhibition System - for those drives that are avoidance in nature.*</li>
+ * <li>Neither Behavioral System - for those drives that are neither approach nor avoidance
+ * in nature.</li>
+ * </ul>
+ * <i>*If a drive is both approach and avoidance in nature, it should be placed within both of
+ * the behavioral systems (using the addDriveToBothSystems or addDrivesToBothSystems methods). Additionally, if the 
+ * drive belongs to both the BIS and BAS, the SYSTEM_GAIN (of the enumerated type TypicalInputs) will be set
+ * by the MS to the average of the gains of the two systems.
+ * <p>
+ * Note that each of the behavioral systems has its own gain parameter that can be used when calculating drive strengths 
+ * for the drives located within each system respectively.
+ * <p>
+ * Below is a list of the primary drives and the behavioral system to which they are affiliated:<br>
+ * <ul>
+ * <li>FOOD - Approach (BAS)</li>
+ * <li>WATER - Approach (BAS)</li>
+ * <li>REST - Avoidance (BIS)</li>
+ * <li>AVOIDING_PHYSICAL_DANGER - Avoidance (BIS)</li>
+ * <li>REPRODUCTION - Approach (BAS)</li>
+ * <li>AVOIDING_THE_UNPLEASANT - Avoidance (BIS)</li>
+ * <li>AFFILIATION_AND_BELONGINGNESS - Both</li>
+ * <li>RECOGNITION_AND_ACHEIVEMENT - Approach (BAS)</li>
+ * <li>DOMINANCE_AND_POWER - Approach (BAS)</li>
+ * <li>AUTONOMY - Both</li>
+ * <li>DEFERENCE - Both</li>
+ * <li>SIMILANCE - Both</li>
+ * <li>FAIRNESS - Both</li>
+ * <li>HONOR - Avoidance (BIS)</li>
+ * <li>NURTURANCE - Approach (BAS)</li>
+ * <li>CONSERVATION - Avoidance (BIS)</li>
+ * <li>CURIOSITY - Approach (BAS)</li>
+ * </ul>
+ * The CLARION Library is perfectly capable of handing drives that are arbitrarily defined (i.e. not a 
+ * primary drive). However, conceptually, any user defined drives should be thought of as a
+ * derived drive that (theoretically, not implementationally) extends from a combination of the
+ * primary drives. For example, the hypothetical drive for "success" could derive from a
+ * combination of the Affiliation & Belongingness, Recognition & Achievement, Dominance & Power, etc.
+ * drives.
+ * <p>
+ * Although, conceptually, the goal structure is also located on the top level of the MS, implementationally 
+ * it is logical to specify it as an intermediate module since it is used by several subsystems. 
+ * Therefore, in the CLARION Library, the goal structure is contained within the CLARION class.
+ * <p>
+ * Note that it is not required that a CLARION agent have an MS at all. If the task being performed
+ * by the agent does not require an MS, the user can choose simply not to attach the MS to the
+ * CLARION agent and that agent will be able to operate quite successfully without this subsystem being
+ * specified (for some tasks).
+ * @version 6.0.6
+ * @author Nick Wilson
+ */
+public final class MS extends AbstractSubsystem{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class contains 15 clusters of instance methods that do not have any accessed class members in common. For example, a cluster might have 2 methods that access only the class fields x and y, and another cluster might have 2 other methods that access only the class fields a and b. By comparison, 99% of the classes in the CodeGuru reference dataset contain fewer clusters. A high number of these clusters indicates low class cohesion. Classes with low class cohesion contain unrelated operations which make them difficult to understand and less likely to be used. We recommend that you simplify this class or break it into multiple cohesive classes.